packages/commons/src/interop-token/models.ts

```diff
@@ -3,134 +3,255 @@ import {
-export const ORGANIZATION_EXTERNAL_ID_ORIGIN_CLAIM = "origin";
-export const ORGANIZATION_EXTERNAL_ID_VALUE_CLAIM = "value";
-export const USER_ROLES = "user-roles";
-const PURPOSE_ID_CLAIM = "purposeId";
-export const ROLE_CLAIM = "role";
-
-export interface InteropJwtHeader {
-  alg: string;
-  use: string;
-  typ: string;
-  kid: string;
-}
-
-export type InteropJwtCommonPayload = {
-  jti: string;
-  iss: string;
-  aud: string[] | string;
-  iat: number;
-  nbf: number;
-  exp: number;
-};
+import {
+  SUPPORT_USER_ID,
+  systemRole,
+  userRole,
+  UserRole,
+} from "../auth/roles.js";
+
+// Type utility to transform a string into an array of strings comma separated
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export const CommaSeparatedStringToArray = <T extends z.ZodType>(t: T) =>
+  z
+    .string()
+    .nonempty()
+    .transform((s: string) => s.split(","))
+    .pipe(z.array(t));
+
+export const InteropJwtHeader = z.object({
+  alg: z.string(),
+  use: z.string(),
+  typ: z.string(),
+  kid: z.string(),
+});
+export type InteropJwtHeader = z.infer<typeof InteropJwtHeader>;
+
+const InteropJwtCommonPayload = z.object({
+  // All standard claims except "sub", which is not present in UI tokens
+  iss: z.string(),
+  aud: z.union([z.array(z.string()), CommaSeparatedStringToArray(z.string())]),
+  exp: z.number(),
+  nbf: z.number(),
+  iat: z.number(),
+  jti: z.string(),
+});
+export type InteropJwtCommonPayload = z.infer<typeof InteropJwtCommonPayload>;
 
 /* ========================================== 
     Interop CONSUMER Token 
   ========================================== */
-export type InteropJwtConsumerPayload = InteropJwtCommonPayload & {
-  client_id: ClientId;
-  sub: ClientId;
-  [PURPOSE_ID_CLAIM]: PurposeId;
-  digest?: ClientAssertionDigest;
-};
+export const InteropJwtConsumerPayload = InteropJwtCommonPayload.merge(
+  z.object({
+    client_id: ClientId,
+    sub: ClientId,
+    purposeId: PurposeId,
+    digest: ClientAssertionDigest.optional(),
+  })
+);
+export type InteropJwtConsumerPayload = z.infer<
+  typeof InteropJwtConsumerPayload
+>;
 
-export type InteropConsumerToken = {
-  header: InteropJwtHeader;
-  payload: InteropJwtConsumerPayload;
-  serialized: string;
-};
+export const InteropConsumerToken = z.object({
+  header: InteropJwtHeader,
+  payload: InteropJwtConsumerPayload,
+  serialized: z.string(),
+});
+export type InteropConsumerToken = z.infer<typeof InteropConsumerToken>;
 
-/* ========================================== 
-    Interop API Token 
-  ========================================== */
-export type InteropJwtApiCommonPayload = InteropJwtCommonPayload & {
-  client_id: ClientId;
-  sub: ClientId;
-  [ORGANIZATION_ID_CLAIM]: TenantId;
-};
+// ==========================================
+//     Interop M2M Token
+// ==========================================
+export const InteropJwtApiCommonPayload = InteropJwtCommonPayload.merge(
```

Comment:
For consistency with the following models
```suggestion
export const InteropJwtApiM2MCommonPayload = InteropJwtCommonPayload.merge(
```

Reply:
Resolving, see https://github.com/pagopa/interop-be-monorepo/pull/1863#discussion_r2091315098